### DISCUSSION TITLE
Criação de canais de voz sob demanda

### DISCUSSION THREAD

**Discussion by FerroEduardo** [Softawii/capivara#49] [Ideas]
- [ ] Definir categoria para criação de canais dinâmicos

# Salas
- [ ] Configurações no canal interno do Voice
- [x] Permitir que o dono da sala configure visibilidade do chat
- [x] Permitir que o dono da sala configure o nome
- [x] Permitir que o dono da sala possa banir/kickar usuários da sala
- [x] Permitir que o dono da sala configure limite de usuários
- [x] Permitir que o dono da sala possa convidar pessoas para a sala
- [x] Permitir transformar canal em permanente
- [x] Permitir criação de text chat

# Futuro
- [ ] Filas de espera para salas? (Útil para Squads) [talvez desnecessário]

**Comment by yaansz** [comment]
Ideia de Tabelas

![voice dynamic drawio](https://user-images.githubusercontent.com/49878392/180663874-ed48a566-6202-4e4b-9816-51d19ad31683.png)

**Comment by yaansz** [comment]
Regras de Negócio:
- O canal principal cria canais temporários
- Os canais temporários são associados a canais de texto!
- Os canais de texto podem ser deletados e as atualizações vão acontecer direto no Voice
- Um canal temporário pode ser feito permanente (mantem as mesmas regras mas não é deletado)
- Um canal tem painel de controle para ser controlado pelo dono ou por um moderador
- Transferência de Líder ocorre quando um chat temporário tem membros, mas o dono saiu! (Ai existe uma mensagem para requisitar um novo lider)
- Chat associado só é visível para pessoas na sala
